backend/src/main/kotlin/cz/loono/backend/LetN.kt

```diff
@@ -0,0 +1,7 @@
+package cz.loono.backend
+
+inline fun <T1, T2, R> let2(t1: T1?, t2: T2?, block: (T1, T2) -> R): R? =
+    if (t1 == null || t2 == null) null else block(t1, t2)
+
+inline fun <T1, T2, T3, R> let3(t1: T1?, t2: T2?, t3: T3?, block: (T1, T2, T3) -> R): R? =
```

Comment:
What are advantages to use this way? In my point of view you hide the construction behind a function and for each object creation you will add if with several conditions. To construct object I expect constructors.

Reply:
This is just a higher order function, there are no constructors here. You likely wanted to comment on the usesite, not declaration site.